### PR TITLE
Hb3a integrate peak

### LINKED ISSUE #14031: Enable peak integration in HB3A GUI
Enable peak integration in HB3A's GUI. 

Also fix the issue with push button 'Clear' in tab 'Calculate UB' and make it more flexible to save and load session file from any file directory.

Add a converter to transform SPICE UB matrix to Mantid UB matrix.  The first line is same, the 2nd line of Mantid UB matrix is the third line of the SPICE matrix.  The third line of Mantid UB is negative to the second line of SPICE UB.

### LINKED ISSUE #14366: Implement an algorithm to integrate peaks for HB3A
The inputs of this algorithm are a MDEventWorkspace containing HB3A experiment data and a peak workspace for the peaks to be integrated.  Removing background and masking are also optional for input.

There can be various algorithms to integrate peaks that are supported by this algorithm.  In this issue, only 1 simple approach is required such that all the signals in the defined region of interest will be integrated.

### PR DESCRIPTION
**Pull request by wdzhou**
Fixes #14366.
Fixes #14031. 

== To test ==
Use Experiment 474, Scan 103 to 235

[Release notes](https://github.com/mantidproject/mantid/blob/hb3a_integrate_peak_14366_14031/docs/source/release/v3.8.0/diffraction.rst) have been updated.

### CODE REVIEW

**Comment by martyngigg** [comment]
@wdzhou Can we move this back to 3.7?

**Comment by wdzhou** [comment]
New requirements: 
1. need to show wavelength along with m1

**Comment by wdzhou** [comment]
To do list:
1. Fix the issue to export Lorenz-corrected peak intensities to fullprof file;
2. Left scan process table to show which k-shift of each scan;
3. Create a k-vector editor

**Comment by wdzhou** [comment]
Bugs and improvements:
1. Add FindUBUsingLatticeParameter (FindUBUsingLatticeParameters-v1.html);
2. Modify checkbox 'SelectAll' in 'merge'-tab to push button;
3. Make peak integration work on exp 474 scan 96-392 (upon already merged via 'add peaks')
4. Add a clipboard to tab-'Export to Fullprof' for the content to be exported.

**Comment by VickieLynch** [comment]
Works with HFIR data and exporting files works correctly :shipit:  This should be featured with a screenshot in release notes with the next ticket.

**Comment by VickieLynch** [comment]
Works with HFIR data and exporting files works correctly :shipit:  This should be featured with a screenshot in release notes with the next ticket.

File: Framework/MDAlgorithms/src/IntegratePeaksCWSD.cpp [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+  double max_distance = -1;
+
+  /*
+  signal_t total_signal = 0.;
+  signal_t total_unmasked_signal = 0;
+  size_t num_masked_det = 0;
+  size_t num_unmasked_det = 0;
+  size_t num_det = 0;
+  int testrunnumber = 13;
+  */
```
Comment (FedeMPouzols) [inline Framework/MDAlgorithms/src/IntegratePeaksCWSD.cpp]: Can this be just removed?

**Comment by VickieLynch** [comment]
:shipit:

**Pull request merged**